--- a/wheezy-stage2
+++ b/wheezy-stage2
@@ -240,6 +240,10 @@
 =mNSj
 -----END PGP PUBLIC KEY BLOCK-----
 EOF1
+echo "deb MIRROR RELEASE COMPONENT"  > /etc/apt/sources.list.d/nightlyfirmware.list
+apt-key add - <<EOF1
+PUBKEY_MARKER
+EOF1
 apt-get update
 apt-get install -y libraspberrypi-bin libraspberrypi0 raspberrypi-bootloader
 echo "dwc_otg.lpm_enable=0 console=ttyAMA0,115200 \
--- a/common
+++ b/common
@@ -107,6 +107,8 @@
   sed -i /etc/apt/sources.list -e "s|^deb [^ ]*|deb http://http.debian.net/debian|"
 fi
 cp /usr/share/doc/raspi-config/sample_profile_d.sh /etc/profile.d/raspi-config.sh
+rm /etc/apt/sources.list.d/nightlyfirmware.list
+apt-key del KEY_ID
 apt-get update
 # Default to a known-good nameserver
 printf "nameserver 8.8.8.8\n" > /etc/resolv.conf
